### PR TITLE
ISPN-5682 Aggregating fields of embedded entities doesn't work

### PR DESCRIPTION
**Pull request by anistor**
Jira: https://issues.jboss.org/browse/ISPN-5682

### CODE REVIEW

**Comment by gustavonalle** [comment]
looking

**Comment by anistor** [comment]
@gustavonalle Just found issues with some tests.

File: query/src/main/java/org/infinispan/query/dsl/embedded/impl/QueryEngine.java [hunk 1]
```diff
@@ -181,6 +183,108 @@ private BaseQuery buildQueryWithAggregations(QueryFactory queryFactory, String j
 
       // validity of query is established at this point, no more checks needed
 
+      boolean foundAggregationsOnRepeatedFields = false;
```
Comment (anistor) [inline query/src/main/java/org/infinispan/query/dsl/embedded/impl/QueryEngine.java]: I plan to extract this blob into a method.

**Comment by anistor** [comment]
Updated.

**Comment by gustavonalle** [comment]
There's only 1 failure left at 
```
testGroupBy4(org.infinispan.all.embeddedquery.QueryDslConditionsTest)  Time elapsed: 0.008 sec  <<< FAILURE!
org.junit.ComparisonFailure: expected:<[Y12]> but was:<[ZZ]>
```

**Comment by anistor** [comment]
Fixed.

**Comment by gustavonalle** [comment]
pulling

**Comment by gustavonalle** [comment]
integrated, thanks!

**Pull request closed without merging**